src/main/java/com/facebook/openwifirrm/modules/Modeler.java

```diff
@@ -340,6 +343,57 @@ private void processData(InputData data) {
 		}
 	}
 
+	/**
+	 * NOTE: if a BSSID does not have a non-obsolete entry, it will be returned
+	 * (i.e., it will not be a key in the returned map).
+	 */
+	public Map<String, WifiScanEntry> getAggregatedWifiScans(long obsoletion_period) {
+		Map<String, WifiScanEntry> aggregatedWifiScans = new HashMap<>();
+		for (String bssid : dataModel.latestWifiScans.keySet()) {
+			// flatten the wifiscan entries and sort in reverse chronological order
+			List<List<WifiScanEntry>> scans = dataModel.latestWifiScans.get(bssid);
+			List<WifiScanEntry> mostRecentToOldest = scans.stream().flatMap(list -> list.stream())
+					.sorted((entry1, entry2) -> {
+						return -Long.signum(entry1.tsf - entry2.tsf);
```

Comment:
I've never seen "signum" used before... I think you just want `Long.compare(a, b)`?

Reply:
if `Long.compare(a, b)` works, then, it might already be the default behavior of `sorted`?